src/ORM/Association/HasMany.php

```diff
@@ -104,6 +147,11 @@ public function saveAssociated(EntityInterface $entity, array $options = [])
         $original = $targetEntities;
         $options['_sourceTable'] = $this->source();
 
+        if ($this->_saveStrategy === self::SAVE_REPLACE) {
+            $updateFields = array_fill_keys(array_keys($properties), null);
```

Comment:
@lorenzo , how would you do to check the nullable flag of the foreign key? I cannot find a simple way to do it. What would you recommend me to do for this case?